pyxrf/model/draw_image_rgb.py

```diff
@@ -0,0 +1,353 @@
+
+class DrawImageRGB(Atom):
+    """
+    This class draws RGB image.
+
+    Attributes
+    ----------
+    fig : object
+        matplotlib Figure
+    file_name : str
+    stat_dict : dict
+        determine which image to show
+    data_dict : dict
+        multiple data sets to plot, such as fit data, or roi data
+    data_dict_keys : list
+    data_opt : int
+        index to show which data is chosen to plot
+    dict_to_plot : dict
+        selected data dict to plot, i.e., fitting data or roi is selected
+    items_in_selected_group : list
+        keys of dict_to_plot
+    scale_opt : str
+        linear or log plot
+    color_opt : str
+        orange or gray plot
+    scaler_norm_dict : dict
+        scaler normalization data, from data_dict
+    scaler_items : list
+        keys of scaler_norm_dict
+    scaler_name_index : int
+        index to select on GUI level
+    scaler_data : None or numpy
+        selected scaler data
+    plot_all : Bool
+        to control plot all of the data or not
+    """
+
+    fig = Typed(Figure)
+    stat_dict = Dict()
+    data_dict = Dict()
+    data_dict_keys = List()
+    data_opt = Int(0)
+    #plot_opt = Int(0)
+    #plot_item = Str()
+    dict_to_plot = Dict()
+    items_in_selected_group = List()
+    scale_opt = Str('Linear')
+    scaler_norm_dict = Dict()
+    scaler_items = List()
+    scaler_name_index = Int()
+    scaler_data = Typed(object)
+    plot_all = Bool(False)
+
+    rgb_name_list = List()
+    index_red = Int(0)
+    index_green = Int(1)
+    index_blue = Int(2)
+
+    def __init__(self):
+        self.fig = plt.figure(figsize=(6,6))
+        self.rgb_name_list = ['R', 'G', 'B']
+
+    def data_dict_update(self, change):
+        """
+        Observer function to be connected to the fileio model
+        in the top-level gui.py startup
+
+        Parameters
+        ----------
+        changed : dict
+            This is the dictionary that gets passed to a function
+            with the @observe decorator
+        """
+        self.data_dict = change['value']
+
+    @observe('data_dict', 'data_dict_keys')
+    def init_plot_status(self, change):
+        # initiate the plotting status once new data is coming
+        self.data_opt = 0
+        self.rgb_name_list = ['R', 'G', 'B']
+        self.index_red = 0
+        self.index_green = 1
+        self.index_blue = 2
+
+        # init of scaler for normalization
+        self.scaler_name_index = 0
+
+        self.data_dict_keys = self.data_dict.keys()
+        logger.info('The following groups are included for 2D image display: {}'.format(self.data_dict_keys))
+
+        scaler_groups = [v for v in self.data_dict.keys() if 'scaler' in v]
+        if len(scaler_groups) > 0:
+            #self.scaler_group_name = scaler_groups[0]
+            self.scaler_norm_dict = self.data_dict[scaler_groups[0]]
+            # for GUI purpose only
+            self.scaler_items = []
+            self.scaler_items = self.scaler_norm_dict.keys()
```

Comment:
you probably want to wrap `self.scalar_norm_dict.keys()` in a `list()` for python 2/3 reasons